packages/backend/src/peripherals/StarknetClient.ts

```diff
@@ -0,0 +1,35 @@
+import { RateLimiter } from '@l2beat/common'
+import { Provider } from 'starknet'
+
+interface StarkNetClientOpts {
+  callsPerMinute?: number
+}
+
+export class StarkNetClient {
+  provider: Provider
+
+  constructor(opts?: StarkNetClientOpts) {
+    this.provider = new Provider({ sequencer: { network: 'mainnet-alpha' } })
+    if (opts?.callsPerMinute) {
+      const rateLimiter = new RateLimiter({
+        callsPerMinute: opts.callsPerMinute,
+      })
+      this.getBlockNumber = rateLimiter.wrap(this.getBlockNumber.bind(this))
+      this.getBlock = rateLimiter.wrap(this.getBlock.bind(this))
+    }
+  }
+
+  async getBlockNumber() {
+    const result = await this.provider.getBlock('latest')
```

Comment:
The other way to go would be to create an instance of `RpcProvider` and use `getBlockNumber` method (ref: https://www.starknetjs.com/docs/API/provider#methods-2) - not sure what are the differences, so I decided to choose a simpler route. Please let me know your thoughts @michalsidzej @sz-piotr

Reply:
It is interesting that the provider doesn't have this method.